docs/adr/adr-0017-finding-accessible-colors-and-creating-design-tokens.md

```diff
@@ -0,0 +1,123 @@
+---
+kind: '📌 Docs/Architecture Decision Records'
+---
+# ADR 0017: Finding accessible colors and creating Design Tokens
+
+🗓️ 2021-04-07 · ✍️ Florian Sanders
+## The context
+We started the Clever Components started as a library of components with no design centralisation and only few shared styles.
+
+This allowed us to build a solid base of essential components quickly.
+
+As more components are being developed and more people contribute, the need for centralised design decisions and ways to enforce design consistency become essential.
+
+## The problem
+
+### Accessibility
+The first issue we wanted to tackle was about the **contrast ratios of some of our most used colours**.
+
+Our blue, green and red colors were all below the 3:1 contrast ratio required for big text and way below the 4.5:1 required for regular text.
+
+These colors are **used in all the components**, especially inside buttons which are found everywhere.
+
+### Consistency & Maintainability
+As we listed all of the colors used accross our components, we found out that there were more than 10 shades of our main colors being used with **no consistency and no specific reason**.
+
+We also found out that **different color code systems were being used with no reason**.
+
+Sometimes the same color could be used with hex code inside a component and hsl in another.
+
+This made even more difficult to change all the colors at once.
+
+### Theming & Customisation
+Since components did not share variables for colors, it was **impossible to modify / adjust them all at once**.
+
+The possibility to **modify all of the colors at once** is the first essential step towards **theme customisation** that would allow for a dark mode for instance.
+
+## The Solution
+
+### Find new accessible colors
+First, we needed to find accessible colors to replace our blue, green, orange and red colors.
+
+#### Criteria
+Colors we selected needed to:
+* reach at least 4.5:1 of contrast ratio when associated with white,
+* match other colors used in the legacy parts of the console,
+* draw attention without standing out too much.
+
+Note: We selected colors that reached **at least 4.5:1 of contrast ratio when associated with white**. This is because we wanted colors that **could be used both with big and regular texts**. Having colors that can only be used with big texts (3:1 of contrast ratio) would have only made things more complicated.
```

Comment:
Maybe an AsciiDoc reflex but I like to write such "note block" with `NOTE: Lorem` (in uppercase)